src/Uno.Toolkit.UI/Controls/Chips/ChipGroup.cs

```diff
@@ -50,32 +50,30 @@ private void OnSelectionMemberPathChanged(DependencyPropertyChangedEventArgs e)
 				: null;
 			foreach (var container in GetItemContainers())
 			{
+				container.ClearValue(Chip.IsCheckedProperty);
 				if (binding != null)
 				{
 					container.SetBinding(Chip.IsCheckedProperty, binding);
 				}
-
-				container.ClearValue(Chip.IsCheckedProperty);
 			}
 		}
 
 		private void ApplyIconTemplate()
 		{
-			if (IconTemplate != null)
+			var itemTemplate = IconTemplate;
+			foreach (var container in GetItemContainers())
 			{
-				foreach (var container in GetItemContainers())
-				{
-					container.Icon = container.Content;
-					container.IconTemplate = IconTemplate;
-				}
+				container.Icon = itemTemplate != null ? container.Content : null;
```

Comment:
There is a null warning here.

Reply:
Will address this in a follow up PR along with runtime tests.